src/utp_stream.cpp

```diff
@@ -341,6 +488,9 @@ struct utp_socket_impl
 
 	void set_state(int s);
 
+	packet *acquire_packet(std::uint16_t allocate) { return m_packet_pool->acquire(allocate); }
```

Comment:
actually, never mind, I don't think that would work.

However, this function should not take a ``uint16_t``. It should either take a ``std::size_t`` or ``int``. that way the callers don't have to cast